src/lib.rs

```diff
@@ -1,23 +1,23 @@
 //! Lazily evaluated monadic IO actions.
 
-use std::{ io::{ Read, Result, Write }, path::Path, mem, fs };
+use std::{ io::{ Read, Result, Write }, path::Path, mem, fs, marker::PhantomData };
 use Inner::{ Eval, Pure, Done };
 
 /// A chain of lazy, composable IO actions.
-pub struct IO <O> (Inner<O>);
+pub struct IO <'a, O = ()> (Inner<'a, O>);
 
-enum Inner <O> {
-    Eval (Box<dyn Action<Output = O>>),
+enum Inner <'a, O> {
+    Eval (Box<dyn Action<'a, Output = O> + 'a>),
```

Comment:
I'd also like to avoid exposing the `Action` trait to the end user because that would make it more difficult to keep the internals as internals. Also, indeed, I want to avoid the psychic damage of introducing *more* parameters to end users (at the cost of redirecting that psychic damage to contributors)